### PR TITLE
chore(rust/gtk-client): allow windows to be closed without exiting the app

### PR DESCRIPTION
**Pull request by ReactorScram**
Also allows windows to be closed and re-opened. Tauri behaved exactly the same way, if you "close" a window it completely destroys it and panics if you ask to show it again.

### CODE REVIEW

File: rust/gtk-client/Cargo.lock [hunk 1]
```diff
@@ -1497,6 +1497,7 @@ dependencies = [
  "rangemap",
  "secrecy",
  "serde",
+ "serde_json",
```
Comment (ReactorScram) [inline rust/gtk-client/Cargo.lock]: I'm not sure how this got here but if I remove it, Cargo puts it back.

File: rust/gtk-client/Cargo.lock
Comment (ReactorScram) [inline rust/gtk-client/Cargo.lock:1]: This PR should not change `Cargo.lock`, I'm not sure why it did.

File: rust/gtk-client/src/main.rs [hunk 2]
```diff
@@ -77,9 +77,10 @@ fn main() -> Result<()> {
     let common::logging::Handles {
         logger: _logger,
         reloader: log_filter_reloader,
-    } = start_logging("debug")?; // TODO
-                                 // The runtime must be multi-thread so that the main thread is free for GTK to consume
-                                 // As long as Tokio has at least 1 worker thread (i.e. there is at least 1 CPU core in the system) this will work.
+    } = start_logging("info")?; // TODO: Load log filter from settings file
```
Comment (ReactorScram) [inline rust/gtk-client/src/main.rs:80]: Dropped this to `info` so it's less noisy for debugging

File: rust/gtk-client/src/main.rs [hunk 3]
```diff
@@ -199,19 +222,42 @@ async fn accept_deep_links(mut server: deep_link::Server, ctlr_tx: CtlrTx) -> Re
...
+/// Waits for the UI to be built and then starts the main thread loop.
+///
+/// This function typically never returns, GLib just stops polling it when the GTK app quits
+async fn run_main_thread_loop(
+    app: gtk::Application,
+    main_rx: mpsc::Receiver<MainThreadReq>,
+    tray_icon: tray_icon::TrayIcon,
+    ui_cell: Rc<RefCell<Option<Ui>>>,
+    mut ui_ready_rx: mpsc::Receiver<()>,
+) -> Result<()> {
```
Comment (ReactorScram) [inline rust/gtk-client/src/main.rs:241]: Extracted so that some things like the `RefCell` can be consumed here

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**